src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -204,19 +204,10 @@ void DoorLockServer::SetUserCommandHandler(chip::app::CommandHandler * commandOb
         return;
     }
 
-    auto fabricIdx = getFabricIndex(commandObj);
-    if (chip::kUndefinedFabricIndex == fabricIdx)
+    auto operationalId = getOperationalId(commandObj);
+    if (operationalId == OperationalId::UndefinedOperationalId())
```

Comment:
This is wrong, in general, no?  I mean, we're not putting door locks in groups so far, but when we do, this will break things.

Given this sort of thing, I'd like to suggest that this PR should be split up into a number of small PRs that might have a chance of being reviewed:

1) Add the new APIs.
2) Incrementally convert consumers to the new APIs, having area experts review the code in their areas.
3) Remove the old APIs.

That's the implementation strategy for this that we discussed weeks ago, so seeing this huge PR is ... surprising.  And the risk involved in something like this, that no one will review carefully without a big time investment, is just not OK at this point.

Reply:
+1 here: introduction of new type for readability/maintainance should not introduce behaviour differences in the same PR.